app/models/user/academics/degree_progress/graduate/student_links.rb

```diff
@@ -0,0 +1,76 @@
+module User
+  module Academics
+    module DegreeProgress
+      module Graduate
+        class StudentLinks
+          attr_reader :user
+
+          APR_LINK_ID_HAAS = 'UC_CX_APR_RPT_GRD_STDNT_HAAS'
+          APR_LINK_ID_LAW = 'UC_CX_APR_RPT_GRD_STDNT_LAW'
+          APR_LINK_ID_GRAD = 'UC_CX_APR_RPT_GRD_STDNT'
+
+          def initialize(user)
+            @user = user
+          end
+
+          def links
+            links = {}
+            links[:academic_progress_report_haas] = academic_progress_report_haas_link if apr_haas_student?
+            links[:academic_progress_report_law] = academic_progress_report_law_link if apr_law_student?
+            links[:academic_progress_report_grad] = academic_progress_report_grad_link if apr_non_haas_grad_student?
+            links
+          end
+
+          def academic_progress_report_haas_link
+            LinkFetcher.fetch_link(APR_LINK_ID_HAAS, { :EMPLID => user.campus_solutions_id })
+          end
+
+          def academic_progress_report_law_link
+            LinkFetcher.fetch_link(APR_LINK_ID_LAW, { :EMPLID => user.campus_solutions_id })
+          end
+
+          def academic_progress_report_grad_link
+            LinkFetcher.fetch_link(APR_LINK_ID_GRAD, { :EMPLID => user.campus_solutions_id })
+          end
```

Comment:
@redconfetti are these links ever presented _to a *student*_?

Reply:
@marcpechaitis - Yes

Reply:
@redconfetti ,
I have done some experiments, and it turns out the value for `&ucEmplid` is ignored when you access the APR page within PeopleSoft.  It turns out the page will only present the data that is associated with the EMPLID value of the user per the PSOPRDEFN table in Campus Solutions.  I am going to recommend (insist) that we remove it from the URL.  

I had a scare that we opened ourselves up to a big security risk by including it in the URL (what happens if the user manipulates the URL to the EMPLID value for another student? will s/he see the other student's APR?).  I think that we are better off completely removing it from the URL, just to be safe.

We will have to make the change to the Link API data on the Campus Solutions side, as well.  I will work with the BSA to ensure that everything still works once we remove EMPLID.

cc: @nkutub , @developish , @frankiepun